src/core/udp_connection.h

```diff
@@ -21,6 +21,8 @@ class UdpConnection : public Connection {
 
     bool send_message(const mavlink_message_t &message);
 
+    void add_remote(const std::string &remote_ip, const int remote_port, const uint8_t remote_sysid = 0);
```

Comment:
If we use this `add_remote` what happens to the `local_ip` and `local_port` that we use in the constructor? I wonder if we could create a remote connection using the remote IP in the constructor but I might be confused about networking.

Reply:
UDP socket is always bound to local_ip and local_port. In case of add_remote the local_ip is localhost and local_port is automatically chosen from one of the free ports.
If we move add_remote to constructor we would have to add a bool flag indicating that it is "remote".